rest/model/admin.go

```diff
@@ -0,0 +1,118 @@
+package model
+
+import (
+	"errors"
+
+	"github.com/evergreen-ci/evergreen/model/admin"
+)
+
+// APIAdminSettings is the structure of a response to the admin route
+type APIAdminSettings struct {
+	Banner       APIString       `json:"banner"`
+	ServiceFlags APIServiceFlags `json:"service_flags"`
+}
+
+// APIBanner is a public structure representing the banner part of the admin settings
+type APIBanner struct {
+	Text APIString `json:"banner"`
+}
+
+// APIServiceFlags is a public structure representing the admin service flags
+type APIServiceFlags struct {
+	TaskDispatchDisabled  bool `json:"task_dispatch_disabled"`
+	HostinitDisabled      bool `json:"hostinit_disabled"`
+	MonitorDisabled       bool `json:"monitor_disabled"`
+	NotificationsDisabled bool `json:"notifications_disabled"`
+	AlertsDisabled        bool `json:"alerts_disabled"`
+	TaskrunnerDisabled    bool `json:"taskrunner_disabled"`
+	RepotrackerDisabled   bool `json:"repotracker_disabled"`
+	SchedulerDisabled     bool `json:"scheduler_disabled"`
+}
+
+// BuildFromService builds a model from the service layer
+func (as *APIAdminSettings) BuildFromService(h interface{}) error {
+	switch v := h.(type) {
+	case *admin.AdminSettings:
+		as.Banner = APIString(v.Banner)
+		as.ServiceFlags = APIServiceFlags{
+			TaskDispatchDisabled:  v.ServiceFlags.TaskDispatchDisabled,
+			HostinitDisabled:      v.ServiceFlags.HostinitDisabled,
+			MonitorDisabled:       v.ServiceFlags.MonitorDisabled,
+			NotificationsDisabled: v.ServiceFlags.NotificationsDisabled,
+			AlertsDisabled:        v.ServiceFlags.AlertsDisabled,
+			TaskrunnerDisabled:    v.ServiceFlags.TaskrunnerDisabled,
+			RepotrackerDisabled:   v.ServiceFlags.RepotrackerDisabled,
+			SchedulerDisabled:     v.ServiceFlags.SchedulerDisabled,
+		}
+	default:
+		return errors.New("Incorrect type when unmarshalling admin settings")
```

Comment:
you can do something like: 

    errors.Errorf("%T is not a supported admin settings type", h)

Reply:
Done